frontend/packages/dev-console/src/components/catalog/utils/catalog-view-utils.ts

```diff
@@ -0,0 +1,74 @@
+import * as _ from 'lodash';
+import { history } from '@console/internal/components/utils';
+import { FilterTypes, getActiveFilters } from './filter-utils';
+
+export const setURLParams = (params) => {
+  const url = new URL(window.location.href);
+  const searchParams = `?${params.toString()}${url.hash}`;
+
+  history.replace(`${url.pathname}${searchParams}`);
+};
+
+export const updateURLParams = (paramName, value) => {
+  const params = new URLSearchParams(window.location.search);
+
+  if (value) {
+    params.set(paramName, Array.isArray(value) ? JSON.stringify(value) : value);
+  } else {
+    params.delete(paramName);
+  }
+  setURLParams(params);
+};
+
+export const clearFilterURLParams = (selectedCategory) => {
+  const params = new URLSearchParams();
+
+  if (selectedCategory) {
+    params.set(FilterTypes.category, selectedCategory);
+  }
+
+  setURLParams(params);
+};
```

Comment:
Works fine, I was a little bit confused why clear settings a category. This would help me, but its not required to change:

```suggestion
export const clearFilterURLParams = (selectedCategory) => {
  const params = new URLSearchParams(window.location.search);

  params.delete(FilterTypes.keyword, selectedCategory);

  setURLParams(params);
};
```

Reply:
This is because we would want to remove all the URL params except the category one. the params can be anything since the filters are dynamic coming from extension. I've moved this logic inside the component now so that it's more clear.

Reply:
Done.